### PR TITLE
creating function to initialize hauls

### PR DESCRIPTION
**Pull request by melinashak-noaa**
Created an initHauls file to initialize empty hauls and compute calculate variables. Set officialTotalCatch here. 

Also modified fish ticket component to read from config and 

Fixed up boatnet-licenses components (this needs to be re-named it's the component used to add and subtract observers and licenses): https://github.com/nwfsc-fram/boatnet/issues/1162

image

Re-factored fish ticket component to read from config, in order to be more flexible and use it for non flow-scale catch: https://github.com/nwfsc-fram/boatnet/issues/1163
image

https://github.com/nwfsc-fram/boatnet/issues/1188
https://github.com/nwfsc-fram/boatnet/issues/1180

### CODE REVIEW

**Comment by melinashak-noaa** [comment]
https://github.com/nwfsc-fram/boatnet/issues/1180

File: apps/ashop/src/data/initHauls.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        if (typeof measurement.value === 'string') {
+            return parseInt(measurement.value, 10);
+        } else {
+            return measurement.value;
+        }
+    }
+    return 0;
+}
+
+function setOfficialTotalCatch(haul: AshopHaul) {
```
Comment (wsmith-nwfsc) [inline apps/ashop/src/data/initHauls.ts]:
```typescript
function setOfficialTotalCatch(haul: AshopHaul): Measurement {
```

File: apps/ashop/src/data/initHauls.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        observerEstCatch += getMeasurementVal(vesselEstMeasurement);
+    }
+    return {
+        measurementType: 'weight',
+        units: 'kg',
+        value: observerEstCatch
+    };
+}
+
+export function initAshopHaul(haulNum: number) {
```
Comment (wsmith-nwfsc) [inline apps/ashop/src/data/initHauls.ts]: add return type as above,
```typescript
export function initAshopHaul(haulNum: number): AshopHaul
```

File: apps/ashop/src/helpers/cruiseInfo.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 import { remove } from 'lodash';
+import { allDocs } from './queries';
 
 let cruise: AshopCruise;
 const db = pouchService.db;
 
 export async function getCruise(type: string, docs: any) {
-    cruise = docs.filter((row: any) => row.doc.type === type + '-cruise');
-    cruise = cruise[0] ? cruise[0].doc : undefined;
+    cruise = docs.filter((row: any) => row.type === type + '-cruise');
```
Comment (wsmith-nwfsc) [inline apps/ashop/src/helpers/cruiseInfo.ts]: This cool, but I think you might want to create a view in Pouch that does this. 
Parsing allDocs may be more of a perf hit that we want.

File: apps/ashop/src/data/initHauls.ts (hunk 1 above)
Comment (melinashak-noaa) [reply apps/ashop/src/data/initHauls.ts]: done

File: apps/ashop/src/data/initHauls.ts (hunk 2 above)
Comment (melinashak-noaa) [reply apps/ashop/src/data/initHauls.ts]: done

File: apps/ashop/src/helpers/cruiseInfo.ts (hunk 3 above)
Comment (melinashak-noaa) [reply apps/ashop/src/helpers/cruiseInfo.ts]: good point, i'll create an issue for this and work on it next: https://github.com/nwfsc-fram/boatnet/issues/1193

Gonna just push this code in for now, trying to get a-shop in prod working and there's some changes in here i want to get out.

**Pull request merged**